cep-????.md

```diff
@@ -0,0 +1,160 @@
+as provided by the generic values in the [`archspec/archspec-json` database](https://github.com/archspec/archspec-json/blob/v0.2.5/cpu/microarchitectures.json), plus some exceptions. For example,  `conda/conda` reports its generic values for Intel/AMD and `m*` names for Apple:
+
+- Generic: `x86_64_v1`, `x86_64_v2`, `x86_64_v3`, `x86_64_v4`, `arm`, `ppc`...
+- Apple: For M1, the value is `m1`. M2 and M3 are reported as `m2` and `m3`, respectively.
+
+If the microarchitecture cannot be detected or the target platform does not match the native platform, the build string MUST be set to the second component of the target platform, mapped with these rules:
+
+| Target platform | Reported `archspec` build string |
+|-----------------|----------------------------------|
+| `*-32` | `x86` |
+| `*-64` | `x86_64` |
+| `*-armv6l` | `armv6l`  |
+| `*-armv7l` | `armv7l`  |
+| `*-aarch64` | `aarch64`  |
+| `*-arm64` | `arm64`  |
+| `*-ppc64` | `ppc64`  |
+| `*-ppc64le` | `ppc64le`  |
+| `*-riscv64` | `riscv64`  |
+| `*-s390x` | `s390x`  |
+| `zos-z` | `0` |
+| Any other value | `0` |
+
+The build string MUST be overridable with the `CONDA_OVERRIDE_ARCHSPEC` environment variable, if set to a non-empty value.
+
+#### `__cuda`
+
+This virtual package MUST be present when the system exhibits GPU drivers compatible with the CUDA runtimes. When available, the version value MUST be set to the oldest CUDA version supported by the detected drivers (i.e. the formatted value of `libcuda.cuDriverGetVersion()`), constrained to the first two components (major and minor) and formatted as `{major}.{minor}`. The build string MUST be `0`.
+
+The version MUST be overridable with the `CONDA_OVERRIDE_CUDA` environment variable, if set to a non-empty value.
+
+#### `__glibc`
+
+This virtual package MUST be present when the native platform is `linux-*`. Its version value MUST be set to the system GNU `libc` version, constrained to the first two components (major and minor) formatted as `{major}.{minor}`. The build string MUST be `0`.
+
+The version MUST be overridable with the `CONDA_OVERRIDE_GLIBC` environment variable, if set to a non-empty value.
+
+If the GNU `libc` version could not be estimated (e.g. the tool is not running on Linux), the tool SHOULD provide a default value (e.g. `2.17`) and inform the user of that choice and its possible overrides; e.g. via `CONDA_OVERRIDE_GLIBC`, a CLI flag or a configuration file. The environment variable MUST be ignored when the target platform is not `linux~-*`.
+
+> The GNU `libc` version can be computed via:
+>
+> - Python's `os.confstr("CS_GNU_LIBC_VERSION")`
+> - `getconf GNU_LIBC_VERSION`
+> - `ldd --version`. Note this only applies to GLIBC distros. In MUSL distros this will return the MUSL version instead. In these systems, you might need to locate the GNU `libc.so` library and call it directly.
```

Comment:
On Alma 8 (using GLIBC) it shows the following

```bash
$ ldd --version
ldd (GNU libc) 2.28
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Written by Roland McGrath and Ulrich Drepper.
```

On Alpine 3 (using MUSL) it shows the following

```bash
$ ldd --version
musl libc (aarch64)
Version 1.2.5
Dynamic Program Loader
Usage: /lib/ld-musl-aarch64.so.1 [options] [--] pathname
```

So the `libc` implementation is being reported in both cases.

Maybe we can just say something like

```suggestion
> - `ldd --version`. Please verify that it references GNU libc or GLIBC. For non-standard installs, using a GLIBC compatibility layer, this may require locating the implementation and directly querying.
```

Am still not sure we should be volunteering to support these non-standard installs in this spec though

Reply:
I'd be fine to rule out or defer musl-based systems, unless someone demonstrates that conda packages work by default with a given glibc-on-alpine shim.